### ISSUE TITLE
[java] ConsecutiveAppendsShouldReuse: False positive within if-statement without curly braces

### ISSUE THREAD

**Issue by cowwoc** [pmd/pmd#4910]
**Affects PMD Version:** 
7.0.0

**Rule:** [ConsecutiveAppendsShouldReuse](https://docs.pmd-code.org/latest/pmd_rules_java_performance.html#consecutiveappendsshouldreuse)

**Description:**
It seems that invoking `StringBuilder.append()` within an if-statement triggers a false-positive of the ConsecutiveAppendsShouldReuse rule. At least, I don't see how the code could be modified to please this rule.

**Code Sample demonstrating the issue:**

```java
StringBuilder message = new StringBuilder();
if (nameOfOther == null)
	message.append(validator.configuration().stringMappers().toString(other));
else
	message.append(MessageBuilder.quoteName(nameOfOther));
message.append('.');
```

**Error Message:**
```
[INFO] PMD Failure: com.github.cowwoc.requirements.java.internal.implementation.message.ComparableMessages:47 Rule:ConsecutiveAppendsShouldReuse Priority:3 StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable..
```

where line 47 points to `message.append(validator.configuration().stringMappers().toString(other));`

**Running PMD through:** *Maven*
maven-pmd-plugin 3.21.2

**Comment by PimvanderLoos** [comment]
Specifically, the issue seems to trigger when omitting the curly braces for both the if-clause and the else-clause.

For example, the `foo` method below will trigger  `ConsecutiveAppendsShouldReuse`. The methods `bar`, `baz`, and `qux` will not trigger the rule.

```java
public static void foo(boolean condition, StringBuilder builder) {
    if (condition)
        builder.append("foo"); // <-- ConsecutiveAppendsShouldReuse at this line
    else
        builder.append("bar");
}

public static void bar(boolean condition, StringBuilder builder) {
    if (condition) {
        builder.append("foo");
    } else {
        builder.append("bar");
    }
}

public static void baz(boolean condition, StringBuilder builder) {
    if (condition)
        builder.append("foo");
    else {
        builder.append("bar");
    }
}

public static void qux(boolean condition, StringBuilder builder) {
    if (condition) {
        builder.append("foo");
    } else
        builder.append("bar");
}
```
